scripts/common/common.sh

```diff
@@ -1036,3 +1036,54 @@ function canonical_image_name() {
     fi
     echo "$image"
 }
+
+# wait_for_running_pods waits for pod(s) in a given namspace to transition to Running status
+function wait_for_running_pods() {
+    local namespace=$1
+    local pods_not_ready=0
+    local ns_pods=
+    local status=
+    local containers=
+
+    ns_pods=$(kubectl get pods -n "$namespace" -o jsonpath='{.items[*].metadata.name}')
+
+    if [ -z "$ns_pods" ]; then
+        log "No pods found in namespace $namespace."
+        return 0
+    fi
+
+    for pod in $ns_pods; do
+        status=$(kubectl get pod "$pod" -n "$namespace" -o jsonpath='{.status.phase}')
+
+        # ignore pods spawned by Jobs
+        if kubectl get pod "$pod" -n "$namespace" -o jsonpath='{.metadata.ownerReferences[*].kind}' | grep -q "Job"; then
+            continue
+        fi
+
+        if [ "$status" != "Running" ]; then
```

Comment:
The provisioner pod will have status completed so that will fail always.

Reply:
Pods with status `Completed     0/1` are managed by the Job controller. I didn't think it was worth checking the status of the Job pods but I'm amenable to change my mind.

Reply:
I am just looking for we ensure that we will not fail because pods are completed.